AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine+IncomingEventReconciliationQueueEvent.swift

```diff
@@ -13,22 +13,32 @@ import Foundation
 extension RemoteSyncEngine {
     @available(iOS 13.0, *)
     func onReceiveCompletion(receiveCompletion: Subscribers.Completion<DataStoreError>) {
-        let semaphore = DispatchSemaphore(value: 1)
-        semaphore.wait()
-        if case .syncEngineActive = stateMachine.state {
-            switch receiveCompletion {
-            case .failure(let error):
-                stateMachine.notify(action: .errored(error))
-            case .finished:
-                stateMachine.notify(action: .errored(nil))
-            }
+        switch stateMachine.state {
+        case .initializeSubscriptions:
+            notifyError(receiveCompletion: receiveCompletion)
+        case .syncEngineActive:
+            notifyError(receiveCompletion: receiveCompletion)
+        default:
+            break
+        }
+    }
+
+    @available(iOS 13.0, *)
```

Comment:
question: if the extensions is marked with `@available(iOS 13.0, *)` does each method also need it?

Reply:
Hmm good call out, looks like it isn't needed!

Reply:
https://github.com/aws-amplify/amplify-ios/pull/329/files